vector/src/main/java/im/vector/app/features/settings/devices/v2/othersessions/OtherSessionsFragment.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.settings.devices.v2.othersessions
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import com.airbnb.mvrx.Success
+import com.airbnb.mvrx.fragmentViewModel
+import com.airbnb.mvrx.withState
+import dagger.hilt.android.AndroidEntryPoint
+import im.vector.app.R
+import im.vector.app.core.platform.VectorBaseBottomSheetDialogFragment
+import im.vector.app.core.platform.VectorBaseBottomSheetDialogFragment.ResultListener.Companion.RESULT_OK
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.core.resources.ColorProvider
+import im.vector.app.databinding.FragmentOtherSessionsBinding
+import im.vector.app.features.settings.devices.v2.DeviceFullInfo
+import im.vector.app.features.settings.devices.v2.DevicesAction
+import im.vector.app.features.settings.devices.v2.DevicesViewModel
+import im.vector.app.features.settings.devices.v2.VectorSettingsDevicesViewNavigator
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterBottomSheet
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterType
+import im.vector.app.features.settings.devices.v2.list.OtherSessionsView
+import im.vector.app.features.settings.devices.v2.list.SESSION_IS_MARKED_AS_INACTIVE_AFTER_DAYS
+import im.vector.app.features.themes.ThemeUtils
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class OtherSessionsFragment :
+        VectorBaseFragment<FragmentOtherSessionsBinding>(),
+        VectorBaseBottomSheetDialogFragment.ResultListener,
+        OtherSessionsView.Callback {
+
+    private val viewModel: DevicesViewModel by fragmentViewModel()
+    @Inject lateinit var colorProvider: ColorProvider
+    @Inject lateinit var viewNavigator: VectorSettingsDevicesViewNavigator
```

Comment:
The idea of this navigator was to be used only in the `VectorSettingsDevicesFragment`. In the same manner we have a `ViewModel`, a `ViewState`, `ViewActions` and `ViewEvents` dedicated to a screen, we should have a `ViewNavigator` per `View` in my opinion. It helps understanding what navigation is possible from a dedicated view.

Reply:
Make sense, done.